From 922e410ebbf3b589385863b70e261e267968b9a4 Mon Sep 17 00:00:00 2001
From: Jake Petroules <jake.petroules@qt.io>
Date: Wed, 19 Jul 2017 16:07:34 -0700
Subject: [PATCH] Fix 32-bit build on macOS

Some customers still need this for interoperability with legacy code.
Let's continue to keep it working in 5.9.x, and then move to
64-bit exclusive features (thus dropping 32-bit entirely) in 5.10.

Task-number: QTBUG-58401
Change-Id: Ibb7200c1885e9caba70439df5f7c86c81b1312b5
Reviewed-by: Gabriel de Dietrich <gabriel.dedietrich@qt.io>
---
 src/plugins/platforms/cocoa/qcocoahelpers.h      | 7 ++++++-
 src/plugins/platforms/cocoa/qcocoahelpers.mm     | 5 +++++
 src/plugins/platforms/cocoa/qnswindowdelegate.mm | 4 ++--
 src/widgets/styles/qmacstyle_mac.mm              | 8 ++++----
 4 files changed, 17 insertions(+), 7 deletions(-)

diff --git a/src/plugins/platforms/cocoa/qcocoahelpers.h b/src/plugins/platforms/cocoa/qcocoahelpers.h
index a2e0876073..4478895538 100644
--- src/plugins/platforms/cocoa/qcocoahelpers.h
+++ src/plugins/platforms/cocoa/qcocoahelpers.h
@@ -167,7 +167,12 @@ QT_END_NAMESPACE
 - (void)onCancelClicked;
 @end
 
-@interface QT_MANGLE_NAMESPACE(QNSPanelContentsWrapper) : NSView
+@interface QT_MANGLE_NAMESPACE(QNSPanelContentsWrapper) : NSView {
+    NSButton *_okButton;
+    NSButton *_cancelButton;
+    NSView *_panelContents;
+    NSEdgeInsets _panelContentsMargins;
+}
 
 @property (nonatomic, readonly) NSButton *okButton;
 @property (nonatomic, readonly) NSButton *cancelButton;
diff --git a/src/plugins/platforms/cocoa/qcocoahelpers.mm b/src/plugins/platforms/cocoa/qcocoahelpers.mm
index 232e40769b..5d0f13c5a9 100644
--- src/plugins/platforms/cocoa/qcocoahelpers.mm
+++ src/plugins/platforms/cocoa/qcocoahelpers.mm
@@ -303,6 +303,11 @@ QString qt_mac_removeAmpersandEscapes(QString s)
  */
 @implementation QNSPanelContentsWrapper
 
+@synthesize okButton = _okButton;
+@synthesize cancelButton = _cancelButton;
+@synthesize panelContents = _panelContents;
+@synthesize panelContentsMargins = _panelContentsMargins;
+
 - (instancetype)initWithPanelDelegate:(id<QT_MANGLE_NAMESPACE(QNSPanelDelegate)>)panelDelegate
 {
     if ((self = [super initWithFrame:NSZeroRect])) {
diff --git a/src/plugins/platforms/cocoa/qnswindowdelegate.mm b/src/plugins/platforms/cocoa/qnswindowdelegate.mm
index 8295d4a36c..1224d138d9 100644
--- src/plugins/platforms/cocoa/qnswindowdelegate.mm
+++ src/plugins/platforms/cocoa/qnswindowdelegate.mm
@@ -75,7 +75,7 @@ - (NSRect)windowWillUseStandardFrame:(NSWindow *)window defaultFrame:(NSRect)new
     // window.screen.visibleFrame directly, as that ensures we have the same
     // behavior for both use-cases/APIs.
     Q_ASSERT(window == m_cocoaWindow->nativeWindow());
-    return m_cocoaWindow->screen()->availableGeometry().toCGRect();
+    return NSRectFromCGRect(m_cocoaWindow->screen()->availableGeometry().toCGRect());
 }
 
 - (BOOL)window:(NSWindow *)window shouldPopUpDocumentPathMenu:(NSMenu *)menu
diff --git a/src/widgets/styles/qmacstyle_mac.mm b/src/widgets/styles/qmacstyle_mac.mm
index 99526683dd..6e8796970f 100644
--- src/widgets/styles/qmacstyle_mac.mm
+++ src/widgets/styles/qmacstyle_mac.mm
@@ -1002,17 +1002,17 @@ static QAquaWidgetSize qt_aqua_guess_size(const QWidget *widg, QSize large, QSiz
             NSBezierPath *focusRingPath;
             if (radius > 0) {
                 const CGFloat roundedRectInset = -1.5;
-                focusRingPath = [NSBezierPath bezierPathWithRoundedRect:CGRectInset(focusRingRect, roundedRectInset, roundedRectInset)
+                focusRingPath = [NSBezierPath bezierPathWithRoundedRect:NSRectFromCGRect(CGRectInset(focusRingRect, roundedRectInset, roundedRectInset))
                                                                 xRadius:radius
                                                                 yRadius:radius];
             } else {
                 const CGFloat outerClipInset = -focusRingWidth / 2;
-                NSBezierPath *focusRingClipPath = [NSBezierPath bezierPathWithRect:CGRectInset(focusRingRect, outerClipInset, outerClipInset)];
+                NSBezierPath *focusRingClipPath = [NSBezierPath bezierPathWithRect:NSRectFromCGRect(CGRectInset(focusRingRect, outerClipInset, outerClipInset))];
                 const CGFloat innerClipInset = 1;
-                NSBezierPath *focusRingInnerClipPath = [NSBezierPath bezierPathWithRect:CGRectInset(focusRingRect, innerClipInset, innerClipInset)];
+                NSBezierPath *focusRingInnerClipPath = [NSBezierPath bezierPathWithRect:NSRectFromCGRect(CGRectInset(focusRingRect, innerClipInset, innerClipInset))];
                 [focusRingClipPath appendBezierPath:focusRingInnerClipPath.bezierPathByReversingPath];
                 [focusRingClipPath setClip];
-                focusRingPath = [NSBezierPath bezierPathWithRect:focusRingRect];
+                focusRingPath = [NSBezierPath bezierPathWithRect:NSRectFromCGRect(focusRingRect)];
                 focusRingPath.lineJoinStyle = NSRoundLineJoinStyle;
             }
 
-- 
2.11.0 (Apple Git-81)

